pandas/tests/indexes/datetimelike.py

```diff
@@ -38,3 +42,190 @@ def test_view(self, indices):
         i_view = i.view(self._holder)
         result = self._holder(i)
         tm.assert_index_equal(result, i_view)
+
+
+class TestDatetimeLikeIndexArithmetic(object):
+    # GH17991 checking for overflows and NaT masking on arithmetic ops
+
+    # TODO: Fill out the matrix of allowed arithmetic operations:
+    #   - __rsub__, __radd__
+    #   - ops with scalars boxed in Index/Series/DataFrame/np.array
+    #   - ops with scalars:
+    #              NaT, Timestamp.min/max, Timedelta.min/max
+    #              datetime, timedelta, date(?),
+    #              relativedelta,
+    #              np.datetime64, np.timedelta64,
+    #              DateOffset,
+    #              Period
+    #   - timezone-aware variants
+    #   - object-dtype, categorical dtype
+    #   - PeriodIndex
+    #   - consistency with .map(...) ?
```

Comment:
have a look thru test_ops, there is lots of coverage for things like this already (or maybe test_base). don't create a giant matrix, rather parametrize as much as possible.

Reply:
I'll take a look.  After adding tests for #7996 (separate branch/PR), this class gets pretty huge.  So yah, parameterization sounds nice.

(Also it looks like tests in this module don't get run, so that needs changing anyway).

Reply:
> (Also it looks like tests in this module don't get run, so that needs changing anyway).

sure they do, classes inherit from this. Pls pls pls don't create a huge matrix of tests w/o looking thru the existing. we cover quite a bit of this already.

Reply:
> Pls pls pls don't create a huge matrix of tests w/o looking thru the existing. we cover quite a bit of this already.

Message received.  Worrying about correctness first, brevity later.